api/client/utils.go

```diff
@@ -94,13 +94,14 @@ func (cli *DockerCli) clientRequest(method, path string, in io.Reader, headers m
 	if resp != nil {
 		serverResp.statusCode = resp.StatusCode
 	}
+
 	if err != nil {
-		if strings.Contains(err.Error(), "connection refused") {
-			return serverResp, errConnectionRefused
+		if strings.Contains(err.Error(), "connection refused") || strings.Contains(err.Error(), "dial unix") || strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
That said, if something creates a new error using the text from the original `net.Error`, this wouldn't work. So we'd need to make sure we'd actually see a `net.Error` here.